client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDriverUri.java

```diff
@@ -477,9 +477,9 @@ private void initCatalogAndSchema()
     private static Properties mergeConnectionProperties(URI uri, Properties driverProperties)
             throws SQLException
     {
-        Map<String, String> defaults = ConnectionProperties.getDefaults();
-        Map<String, String> urlProperties = parseParameters(uri.getQuery());
-        Map<String, String> suppliedProperties = Maps.fromProperties(driverProperties);
+        Map<String, Object> defaults = ConnectionProperties.getDefaults();
+        Map<String, Object> urlProperties = parseParameters(uri.getQuery());
+        Map<String, Object> suppliedProperties = driverProperties.entrySet().stream().collect(toImmutableMap(entry -> (String) entry.getKey(), Entry::getValue));
```

Comment:
nit: Post `.stream` can we move the collect to a new line ?

Reply:
refactored